### PR TITLE
Enable bridge test

### PR DESCRIPTION
**Pull request by vyadavmsft**
This patch enable testing for Bridge mode networking. enabled in this PR: https://github.com/cloud-hypervisor/libvirt/pull/93

Signed-off-by: Vivek Yadav vyadav@microsoft.com

### CODE REVIEW

**Comment by bryteise** [comment]
@vyadavmsft congratulations on getting this working!

**Comment by vyadavmsft** [comment]
thanks, @bryteise   i will squash all these commits to one and push it soon.

File: ch_integration_tests/run_tests.sh [hunk 1]
```diff
@@ -57,5 +57,6 @@ if [ -d "$LINUX_CUSTOM_DIR" ]; then
 fi
 
 pushd ch_integration_tests
-cargo test -- --test-threads=1
+#cargo test -- --test-threads=1
+cargo test -- test_bridge_networking --nocapture RUST_BACKTRACE=1
```
Comment (bryteise) [inline ch_integration_tests/run_tests.sh]: This is a leftover from debugging I think.

Comment (vyadavmsft) [reply ch_integration_tests/run_tests.sh]: oh yes :) let me fix.

File: ch_integration_tests/tests/integration.rs [hunk 2]
```diff
@@ -718,4 +718,96 @@ mod tests {
...
+
+        //configure host IP  since we cant do it from XML for bridge network.
+        Command::new("sh")
+                .arg("-c")
+                .arg("sudo ip addr add 192.168.1.1/24 dev mybr0 ; sudo ip addr add 192.168.2.1/24 dev mybr0")
+                .spawn()
+                .expect("Failed to set IP address.");
+
+        let mut guest_name: [String; 2] = ["guest1".to_string(), "guest2".to_string()];
+        let mut guest_ip: [String; 2] = ["ip1".to_string(), "ip2".to_string()];
```
Comment (bryteise) [inline ch_integration_tests/tests/integration.rs]: I recommend setting these to `Default::default()` instead (handy thing setting lots of initial values).

File: ch_integration_tests/tests/integration.rs [hunk 3]
```diff
@@ -718,4 +718,96 @@ mod tests {
...
+                guest.wait_vm_boot(None).unwrap();
+            });
+
+            let guestname = &guest.vm_name;
+            guest_name[i] = guestname.to_string();
+            assert!(r.is_ok());
+
+            guest_ip[i] = guest
+                .ssh_command(
+                    "ip route get 8.8.8.8 | sed -n '/src/{s/.*src *\\([^ ]*\\).*/\\1/p;q}'",
```
Comment (bryteise) [inline ch_integration_tests/tests/integration.rs:770]: Ah that's a nice way to get an ip without mask of an interface in use!

**Comment by bryteise** [comment]
This looks really nice!

File: ch_integration_tests/tests/integration.rs (hunk 3 above)
Comment (vyadavmsft) [reply ch_integration_tests/tests/integration.rs:770]: Thanks :)

**Comment by bryteise** [comment]
Would you mind squashing the use default initializer commit since it also has the changes to the cargo test line? After that I'm looking to merge this PR.

**Pull request merged**